### PR TITLE
fix #7544 lit and range ambiguous

### LINKED ISSUE #7544: Cannot overload by range
I tried to define:

```nim
type Num = object
#type Natural = range[0 .. high(int)]
type Negative = range[low(int) .. -1]
proc `^`(a: Num, b: static[Natural]): Num = ...
proc `^`(a: Num, b: static[Negative]): Num = ...
```

When using the operator it says both matches for literal -3.

### PR DESCRIPTION
**Pull request by bung87**
fix #7544

### CODE REVIEW

File: tests/generics/t7544.nim [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+#type Natural = range[0 .. high(int)]
+type Negative = range[low(int) .. -1]
+proc `^`(a: bool, b: static[Natural]): int = 1
+proc `^`(a: bool, b: static[Negative]): int = 2
+discard true ^ -3
```
Comment (juancarlospaco) [inline tests/generics/t7544.nim]: At least assert something in the test ?. 🤔

**Comment by bung87** [comment]
well,CI fails shows the mismatch error will cover range check error. as the test case provide index as intlit

**Comment by bung87** [comment]
CI fails unrelated `tests\gc\gcbench.exe: Permission denied`

**Comment by Araq** [comment]
Here is how I know that this doesn't make sense: The manual does not mention the notion of an `indexTypesMatch`.

**Comment by bung87** [comment]
> Here is how I know that this doesn't make sense: The manual does not mention the notion of an `indexTypesMatch`.

so the issue is valid or should close?

**Comment by Araq** [comment]
I don't know, read the manual :P

**Pull request closed without merging**